src/docs/asciidoc/reference/architecture/code-structure.adoc

```diff
@@ -1,2 +1,150 @@
+<3> The `worker` folder contains your business logic.
+<4> The `front` folder contains the front part of your application.
+<5> The `application.json` file lets you configure the logic of your application. (Token, API key, Mailjet configuration, etc...).
+<6> The `.zetarc` file contains the application and account informations (developer login, developer password, appName, etc...). It should not be commited (it contains login and password) so by default it is added to the `.gitignore` file. 
+
+[#architecture_files_tree]
+== Files tree
+
+As seen in the last section, each file has a specific role. We will explain each of them step by step:
+
+[#role_package_json]
+=== package.json
+
+A *ZetaPush application* is based of the `JavaScript ecosystem`.
+So each application has his `package.json` file.
+
+include::{docdir}/common/code/minimal-package-json.adoc[]
+
+The `main` property define the path of the worker entry point. That means that when you run or push your *ZetaPush application*, this is from this entry point that your business logic is launched.
+
+.Location of the entry point
+[NOTE]
+====
+At the present time, the worker code needs to be in the `worker` folder, but you can specify with the `main` property any file you want in the `worker` folder to be the entry point.
+====
+
+[#role_tsconfig]
+=== tsconfig.json
+
+The `tsconfig.json` file is here because by default ZetaPush uses the `TypeScript ecosystem`. For example, the _cloud services_ are written in `TypeScript` and the `worker` folder contains by default a `TypeScript` file.
+You can see <<reference#javascript_vs_typescript, why we use TypeScript>>.
+
+In the `TypeScript ecosystem`, the `tsconfig.json` file is needed to define the root of the application. So in our case, the `tsconfig.json` file is placed at the root of the *ZetaPush application*.
+
+If you are not familiar with `TypeScript`, you can see more informations link:{typescript-url}[here].
+
+In addition to define the root of our application, the `tsconfig.json` file define how the transpilation from `TypeScript` to `JavaScript` is done.
+You can define a lot of configurations according to the link:{typescript-conf-url}[TypeScript documentation].
+
+For the proper functioning of the *ZetaPush application*, the `tsconfig.json` file needs to have some minimal configuration like explained below :
+
+include::{docdir}/common/code/minimal-tsconfig-needed.adoc[]
+
+.Customize my tsconfig.json
+
+In this case of you need to customize your `tsconfig.json` file, for example to use classes only declared in ES6 (es2015), you can add this in the `compilerOptions`. Of course, you can add any properties provided by link:{typescript-conf-url}[TypeScript].
```

Comment:
for example to use classes only declared in ES6 (es2015) -> this is not clear I think